### PR TITLE
Cwl

### PR DESCRIPTION
**Pull request by jimmyv9**
### Minor changes made
-  Anyone should be able to run SJARACNe on their machines by running:
  - `python setup.py build`
  - `python setup.py install`.
- Quality control of input files was added through an extra "step 0" in the workflow
- Answerkey modified in order to pass Travis tests, as Linux and Mac seemingly give different answers.
  - This means `python -m pytest` will not work locally on Mac

### Deletions
- sjaracne.exe (executable to be created by own compiler when building)
- Previous accidental additions and Travis testing

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,17 +7,15 @@ before_install:
...
+  - python setup.py build
 
 install:
   - pip install -r requirements.txt
   - python setup.py install
 
 before_script:
   - eggpath='/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/SJARACNe-0.2.0-py3.6.egg/SJARACNe/bin'
-  - chmod 744 $eggpath/ch_line_ending.py $eggpath/create_consensus_network.py $eggpath/sjaracne.exe
+  - chmod 744 $eggpath/ch_line_ending.py $eggpath/create_consensus_network.py $eggpath/QC_input.py $eggpath/sjaracne.exe
```
Comment (pNKz7mD0jz89KSShpK04tgEKc6OtEInKODtyxWB) [inline .travis.yml]: Suggest changing to 755 which allows executing

File: SJARACNe/bin/QC_input.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        #process rest of the file, making sure tabs are splitting entries
+        for line in fin:
+            words = line.split('\t')
+            total_genes += 1 #add one gene to the total count per line of the exp file
+            for word in words:
+                if ' ' in word:
+                    logging.info("Word with spaces is: {}".format(word))
+                    sys.exit('Error - spaces are not allowed, only tabs can delimit input file. Space found in line {}'.format(total_genes+1))
+                if word.count('.') > 1 and word.isnumeric():
+                    print("Numeric entry missing spacing is: {}".format(word))
```
Comment (adamdingliang) [inline SJARACNe/bin/QC_input.py]: This should go to stderr. Use logging.info()

Comment (pNKz7mD0jz89KSShpK04tgEKc6OtEInKODtyxWB) [inline SJARACNe/bin/QC_input.py]: This should go to stderr. Use logging.info()

File: SJARACNe/bin/QC_input.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                    sys.exit('Error - There are some numeric entries missing tab-spacing in line {}'.format(total_genes+1))
+    logging.info("Number of genes in expression matrix: {}".format(total_genes))
```
Comment (adamdingliang) [inline SJARACNe/bin/QC_input.py]: Can we add one more check to ensure that the number of tokens of every line is identical to the number of tokens in the first line? This is to ensure that the end of the file does not get truncated.

Comment (pNKz7mD0jz89KSShpK04tgEKc6OtEInKODtyxWB) [inline SJARACNe/bin/QC_input.py:41]: Can we add one more check to ensure that the number of tokens of every line is identical to the number of tokens in the first line? This is to ensure that the end of the file does not get truncated.

File: setup.py [hunk 4]
```diff
@@ -1,4 +1,15 @@
...
 from setuptools import setup, find_packages
+from distutils.command.install import install
+from distutils.command.build import build
+from subprocess import call
+
+class Build(build):
+    def run(self):
+        cmd = ["make", "-C", "./SJARACNe"]
+        if call(cmd) != 0:
+            sys.exit(-1)
```
Comment (adamdingliang) [inline setup.py]: If make fails, would this show the error message?

Comment (pNKz7mD0jz89KSShpK04tgEKc6OtEInKODtyxWB) [inline setup.py]: If make fails, would this show the error message?

File: SJARACNe/bin/QC_input.py (hunk 2 above)
Comment (jimmyv9) [reply SJARACNe/bin/QC_input.py]: Overlooked that one, will change it.

Comment (bnL1x4EYnKwARrdJDH00h4a0KD77X67N054drTc) [reply SJARACNe/bin/QC_input.py]: Overlooked that one, will change it.

File: SJARACNe/bin/QC_input.py (hunk 3 above)
Comment (jimmyv9) [reply SJARACNe/bin/QC_input.py]: Yes, we can add that.

Comment (bnL1x4EYnKwARrdJDH00h4a0KD77X67N054drTc) [reply SJARACNe/bin/QC_input.py:41]: Yes, we can add that.

File: setup.py (hunk 4 above)
Comment (jimmyv9) [reply setup.py]: It does show error messages (I just tried running it without a Makefile to double check). Python calls on `make -C ./SJARACNe` on the command line, so any issue with make when running `python setup.py build` is printed on the command line.

Comment (bnL1x4EYnKwARrdJDH00h4a0KD77X67N054drTc) [reply setup.py]: It does show error messages (I just tried running it without a Makefile to double check). Python calls on `make -C ./SJARACNe` on the command line, so any issue with make when running `python setup.py build` is printed on the command line.

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jimmyv9) [reply .travis.yml]: Fixed

Comment (bnL1x4EYnKwARrdJDH00h4a0KD77X67N054drTc) [reply .travis.yml]: Fixed

**Pull request merged**